src/proto/types.rs

```diff
@@ -294,6 +291,15 @@ impl From<Payload> for Bytes {
     }
 }
 
+impl From<Payload> for BytesMut {
+    fn from(value: Payload) -> Self {
+        match value.data.into_inner() {
+            PayloadStorage::Unique(p) => p,
+            PayloadStorage::Shared(p) => p.into(),
```

Comment:
Might want to publicly document this allocation. The `PayloadStorage::Shared(bytes) => bytes.try_into_mut().is_err()` invariant should probably also be documented.

Reply:
Should be a bit better now